lambdas/build-proven-user-identity-details/src/test/java/uk/gov/di/ipv/core/buildprovenuseridentitydetails/BuildProvenUserIdentityDetailsHandlerTest.java

```diff
@@ -577,12 +551,14 @@ private VcStoreItem createVcStoreItem(
         return vcStoreItem;
     }
 
-    private APIGatewayProxyResponseEvent makeRequest(
-            APIGatewayProxyRequestEvent event, Context context) {
-        final var requestType = new TypeReference<Map<String, Object>>() {};
-        final var request = objectMapper.convertValue(event, requestType);
-        final var response = underTest.handleRequest(request, context);
+    private static String getJsonResponse(Map<String, Object> response)
+            throws JsonProcessingException {
+        return objectMapper.writeValueAsString(response);
+    }
 
-        return objectMapper.convertValue(response, APIGatewayProxyResponseEvent.class);
+    private String makeRequest(JourneyRequest event, Context context)
```

Comment:
As an alternative the following may simplify the code which is currently in use in the CheckExistingIdentityHandlerTest:
```java
private <T extends BaseResponse> T handleRequest(
        JourneyRequest request, Context context, Class<T> classType) throws IOException {
    try (var inputStream = new ByteArrayInputStream(mapper.writeValueAsString(request).getBytes());
            var outputStream = new ByteArrayOutputStream()) {
        userTest.handleRequest(inputStream, outputStream, context);
        return mapper.readValue(outputStream.toString(), classType);
    }
}
```

Which can be called like the following.
```java
var journeyResponse = handleRequest(event, context, JourneyResponse.class); // automatically returns a JourneyResponse
```

or:
```java
var journeyResponse = handleRequest(event, context, JourneyErrorResponse.class); // automatically returns a JourneyErrorResponse
```

Reply:
Done